### PR TITLE
MVP to render the cross model relations and offers

### PR DESCRIPTION
**Pull request by hatched**
## Done

As the data and designs are in flux for this feature this is an MVP to get the CMR and Offers rendered to the details pages.

## QA

- Follow [cmr-scenario-1 docs](https://juju.is/docs/cmr-scenario-1) to setup an environment with cross model relations
- Follow [developing while connected to a Juju controller](https://github.com/canonical-web-and-design/jaas-dashboard#developing-while-connected-to-a-juju-controller) to connect your local running instance of the dashboard to the remove environment created above.

## Details

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1215

## Screenshots
Screen Shot 2020-09-24 at 3 19 02 PM
Screen Shot 2020-09-24 at 3 28 55 PM
Screen Shot 2020-09-24 at 3 30 45 PM
Screen Shot 2020-09-24 at 3 31 01 PM

### CODE REVIEW

**Comment by spencerbygraves** [comment]
@hatched is it possible to align the columns a bit better please?

For example:

- Status, Workload, State, Requirer
- Version, Agent, AZ, Interface, Endpoints
- Store, Public address, Instance ID, Type
- OS, Message, Connections

Thanks

image

**Comment by hatched** [comment]
Thanks @spencerbygraves I've created a [follow-up issue](https://github.com/canonical-web-and-design/jaas-dashboard/issues/700) to address that as it's a bit of work and out of scope for this PR. Mind taking another look at the review?

File: src/pages/Models/Details/generators.js [hunk 1]
```diff
@@ -379,3 +391,77 @@ export function generateRelationRows(modelStatusData, baseAppURL) {
...
+      columns: [
+        {
+          content: (
+            <>
+              {generateRelationIconImage(
+                offer.applicationName,
+                modelStatusData,
+                baseAppURL
+              )}
+              {offer.applicationName}:db
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply src/pages/Models/Details/generators.js:413]: This will render duplicate charms but that is ok for this pass

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Code looks good, one hardcoded issue.

**Review by anthonydillon — APPROVED** [review]
QA is good just the hardcoded issue in the code.

**Pull request merged**